core/includes/file.inc

```diff
@@ -456,7 +456,7 @@ function file_prepare_directory(&$directory, $options = FILE_MODIFY_PERMISSIONS)
     if (($options & FILE_CREATE_DIRECTORY) && @backdrop_mkdir($directory, NULL, TRUE)) {
       return backdrop_chmod($directory);
     }
-    return FALSE;
+    return is_dir($directory);
```

Comment:
If I get it right, this might return "true", but doesn't check if it's writable - please correct me if I'm wrong.

Reply:
Good point. PR updated to check for writable dir